packages/hash/frontend/src/pages/type-editor/index.page.tsx

```diff
@@ -0,0 +1,76 @@
+import { useCallback, useEffect, useState } from "react";
+import { useRouter } from "next/router";
+import { Container } from "@mui/material";
+
+import { useUser } from "../../components/hooks/useUser";
+import { NextPageWithLayout } from "../../shared/layout";
+import { useBlockProtocolFunctionsWithOntology } from "./blockprotocol-ontology-functions-hook";
+import {
+  AggregateDataTypesMessageCallback,
+  AggregateEntityTypesMessageCallback,
+  AggregateLinkTypesMessageCallback,
+  AggregatePropertyTypesMessageCallback,
+} from "../../components/hooks/blockProtocolFunctions/ontology/ontology-types-shim";
+
+const Page: NextPageWithLayout = () => {
+  const router = useRouter();
+  const { user, loading, kratosSession } = useUser();
+  const [content, setContent] = useState<string>();
+
+  useEffect(() => {
+    if (loading) {
+      return;
+    }
+
+    if (!kratosSession && !user) {
+      void router.push("/login");
+    }
+  }, [loading, router, user, kratosSession]);
+
+  /** @todo use the User's accountId here. */
+  const functions = useBlockProtocolFunctionsWithOntology("..");
+
+  const getType = useCallback(
```

Comment:
Can you structure this file a bit more clearly to show what's part of the template/boilerplate and what's there for demonstration purposes? (Even just a comment delineating the sections or something)

Reply:
Addressed in 2a01ba68158085cac0e00a2705d8f1f105c21ed4
Added a button to create a property type as well in the demo page

Reply:
Better, thank you!